### PR TITLE
PLT-257: vSphere cloud account implementation.

### PR DESCRIPTION
**Pull request by nikchern**

### CODE REVIEW

File: spectrocloud/resource_cloud_account_vsphere.go [hunk 1]
```diff
@@ -83,26 +98,34 @@ func resourceCloudAccountVsphereRead(_ context.Context, d *schema.ResourceData,
 		return diags
 	}
 
+	diagnostics, done := flattenVsphereCloudAccount(d, account)
+	if done {
+		return diagnostics
+	}
+
+	// Don't read the password!!
```
Comment (TylerGillson) [inline spectrocloud/resource_cloud_account_vsphere.go:106]: consider removing this

File: pkg/client/account_vsphere.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+func (h *V1Client) GetCloudAccountVsphere(uid string) (*models.V1VsphereAccount, error) {
+	client, err := h.GetClusterClient()
+	if err != nil {
+		return nil, err
+	}
+
+	getParams := clusterC.NewV1CloudAccountsVsphereGetParamsWithContext(h.Ctx).WithUID(uid)
+	success, err := client.V1CloudAccountsVsphereGet(getParams)
+	if e, ok := err.(*hapitransport.TransportError); ok && e.HttpCode == 404 {
+		// TODO(saamalik) check with team if this is proper?
```
Comment (TylerGillson) [inline pkg/client/account_vsphere.go:118]: consider removing

File: pkg/client/account_aws.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+func (h *V1Client) GetCloudAccountAws(uid string) (*models.V1AwsAccount, error) {
+	client, err := h.GetClusterClient()
+	if err != nil {
+		return nil, err
+	}
+
+	params := clusterC.NewV1CloudAccountsAwsGetParamsWithContext(h.Ctx).WithUID(uid)
+	success, err := client.V1CloudAccountsAwsGet(params)
+	if e, ok := err.(*hapitransport.TransportError); ok && e.HttpCode == 404 {
+		// TODO(saamalik) check with team if this is proper?
```
Comment (TylerGillson) [inline pkg/client/account_aws.go:94]: consider removing

File: pkg/client/client.go [hunk 4]
```diff
@@ -48,7 +49,7 @@ type V1Client struct {
 	password       string
 	apikey         string
 	transportDebug bool
-	retryAttempts  int
+	RetryAttempts  int
```
Comment (TylerGillson) [inline pkg/client/client.go:52]: any particular reason to export this?

File: pkg/client/addon_deployment_update.go [hunk 5]
```diff
@@ -13,18 +13,18 @@ func (h *V1Client) UpdateAddonDeployment(cluster *models.V1SpectroCluster, body
...
 
 	resolveNotification := true
 	params := clusterC.NewV1SpectroClustersPatchProfilesParamsWithContext(h.Ctx).WithUID(uid).WithBody(body).WithResolveNotification(&resolveNotification)
-	err := patchWithRetry(h, params)
+	err := PatchWithRetry(h, params)
 	return err
 }
 
-func isProfileAttachedByName(cluster *models.V1SpectroCluster, newProfile *models.V1ClusterProfile) (bool, string) {
+func IsProfileAttachedByName(cluster *models.V1SpectroCluster, newProfile *models.V1ClusterProfile) (bool, string) {
```
Comment (TylerGillson) [inline pkg/client/addon_deployment_update.go:27]: any particular reason to export this?

File: pkg/client/addon_deployment_update.go [hunk 6]
```diff
@@ -38,16 +38,16 @@ func isProfileAttachedByName(cluster *models.V1SpectroCluster, newProfile *model
 func (h *V1Client) CreateAddonDeployment(uid string, body *models.V1SpectroClusterProfiles) error {
 	resolveNotification := false // during initial creation we never need to resolve packs.
 	params := clusterC.NewV1SpectroClustersPatchProfilesParamsWithContext(h.Ctx).WithUID(uid).WithBody(body).WithResolveNotification(&resolveNotification)
-	err := patchWithRetry(h, params)
+	err := PatchWithRetry(h, params)
 	return err
 }
 
-func patchWithRetry(h *V1Client, params *clusterC.V1SpectroClustersPatchProfilesParams) error {
+func PatchWithRetry(h *V1Client, params *clusterC.V1SpectroClustersPatchProfilesParams) error {
```
Comment (TylerGillson) [inline pkg/client/addon_deployment_update.go:45]: why export?

File: pkg/client/client.go [hunk 7]
```diff
@@ -74,11 +75,11 @@ type V1Client struct {
 	GetClusterProfileFn     func(string) (*models.V1ClusterProfile, error)
 	DeleteClusterProfileFn  func(string) error
 	// special function for nested mock
-	v1ClusterProfilesDeleteFn            func(params *clusterC.V1ClusterProfilesDeleteParams) (*clusterC.V1ClusterProfilesDeleteNoContent, error)
-	v1ClusterProfilesUIDMetadataUpdateFn func(params *clusterC.V1ClusterProfilesUIDMetadataUpdateParams) (*clusterC.V1ClusterProfilesUIDMetadataUpdateNoContent, error)
-	v1ClusterProfilesUpdateFn            func(params *clusterC.V1ClusterProfilesUpdateParams) (*clusterC.V1ClusterProfilesUpdateNoContent, error)
-	v1ClusterProfilesCreateFn            func(params *clusterC.V1ClusterProfilesCreateParams) (*clusterC.V1ClusterProfilesCreateCreated, error)
-	v1ClusterProfilesPublishFn           func(params *clusterC.V1ClusterProfilesPublishParams) (*models.V1ClusterProfile, error)
+	V1ClusterProfilesDeleteFn            func(params *clusterC.V1ClusterProfilesDeleteParams) (*clusterC.V1ClusterProfilesDeleteNoContent, error)
```
Comment (TylerGillson) [inline pkg/client/client.go:78]: why export these?

File: pkg/client/addon_deployment_update.go (hunk 5 above)
Comment (nikchern) [reply pkg/client/addon_deployment_update.go:27]: this is so unit tests can be in a different module.

File: spectrocloud/resource_cloud_account_vsphere.go (hunk 1 above)
Comment (nikchern) [reply spectrocloud/resource_cloud_account_vsphere.go]: removed.

File: pkg/client/account_aws.go (hunk 3 above)
Comment (nikchern) [reply pkg/client/account_aws.go]: removed.

**Review by TylerGillson — APPROVED** [review]
We should consider refactoring in the future to put unit tests in the same module as the code they test.

**Pull request merged**